Tests/Marketplace/search_and_install_packs.py

```diff
@@ -531,268 +721,282 @@ def search_pack_and_its_dependencies(client: demisto_client,
-        multithreading (bool): Whether to use multithreading to install packs in parallel.
-            If multithreading is used, installation requests will be sent in batches of each pack and its dependencies.
         production_bucket (bool): Whether the installation is in post update mode. Defaults to False.
     Returns (list, bool):
-        A list of the installed packs' ids, or an empty list if is_nightly == True.
+        A list of the installed packs' ids.
         A flag that indicates if the operation succeeded or not.
     """
     host = hostname or client.api_client.configuration.host
-
-    logging.info(f'Starting search for packs to install on: {host}')
-
-    packs_to_install: list = []  # Packs we want to install, to avoid duplications
-    installation_request_body: list = []  # Packs to install, in the request format
-    batch_packs_install_request_body: list = []  # List of lists of packs to install if not using multithreading .
-    # Each list contain one pack and its dependencies.
-    collected_dependencies: list = []  # List of packs that are already in the list to install.
-    master_commit_hash = get_env_var("LAST_UPLOAD_COMMIT")
-
-    lock = Lock()
-
-    kwargs = {
-        'client': client,
-        'packs_to_install': packs_to_install,
-        'installation_request_body': installation_request_body,
-        'lock': lock,
-        'collected_dependencies': collected_dependencies,
-        'production_bucket': production_bucket,
-        'multithreading': multithreading,
-        'batch_packs_install_request_body': batch_packs_install_request_body,
-        'commit_hash': master_commit_hash,
-    }
-
-    if not multithreading:
-        for pack_id in pack_ids:
-            search_pack_and_its_dependencies(pack_id=pack_id, **kwargs)
-
+    commit_hash: str = get_env_var("LAST_UPLOAD_COMMIT")
+
+    logging.info(f"Starting to search and install packs on server {host}")
+    start_time = datetime.utcnow()
+    success = True
+    all_packs_dependencies = {}
+    with ThreadPoolExecutor(max_workers=MAX_WORKERS if multithreading else 1) as pool:
+        futures = [
+            pool.submit(
+                get_pack_and_its_dependencies,
+                pack_id=pack_id,
+                client=client,
+                production_bucket=production_bucket,
+                commit_hash=commit_hash,
+            )
+            for pack_id in pack_ids
+        ]
+
+        for future in as_completed(futures):
+            try:
+                pack_id, dependencies_for_packs = future.result()
+                if dependencies_for_packs is not None:
+                    all_packs_dependencies[pack_id] = dependencies_for_packs
+                else:
+                    logging.error(
+                        f"Failed to search for dependencies of pack '{pack_id}'"
+                    )
+                    success = False
+            except Exception:  # noqa E722
+                logging.error(
```

Comment:
No need to log the details of the exception that was raised?
At least make it a different log message than the one just above (line 956)